studio/src/store.js

```diff
@@ -97,105 +109,94 @@ export function navigateToPage(value) {
+ * Links a store to the URL hash
+ * @param {ReactiveStore} store - The store to link
+ * @param {string|string[]} keys - The key(s) to link
+ * @param {any} defaultValue - The default value to use if the key is not in the hash
  */
-export function linkStoreToHash(store, params, defaultValue) {
-    if (store.hasMeta('hashLink')) {
-        console.error('Cannot link to hash a store that is already linked.');
-        return;
-    }
+export function linkStoreToHash(store, keys, defaultValue) {
+    if (!store) return;
+    const keysArray = Array.isArray(keys) ? keys : [keys];
 
-    const isPrimitive = !Array.isArray(params);
+    const hash = window.location.hash.slice(1);
+    const params = new URLSearchParams(hash);
 
-    function syncFromHash() {
-        const value = store.get();
-        const defaultValues = isPrimitive
-            ? { [params]: defaultValue }
-            : defaultValue || {};
+    let shouldUpdateStore = false;
+    const updates = {};
 
-        if (isPrimitive) {
-            const hashValue = getHashParam(params);
-            if (hashValue && hashValue !== value) {
-                store.set(hashValue);
+    for (const key of keysArray) {
+        if (params.has(key)) {
+            let value = params.get(key);
+            try {
+                value = JSON.parse(value);
+            } catch (e) {
+                // Not JSON, use as is
             }
-        } else {
-            let hasChanges = false;
-            const hashValue = {};
-            for (const param of params) {
-                const paramValue = getHashParam(param);
-                // Handle array values by splitting on commas if the default value is an array
-                hashValue[param] = Array.isArray(defaultValues[param])
-                    ? paramValue
-                        ? paramValue.split(',')
-                        : []
-                    : paramValue;
-
-                if (!looseEquals(hashValue[param], value[param])) {
-                    if (
-                        (!hashValue[param] ||
-                            looseEquals(
-                                hashValue[param],
-                                defaultValues[param],
-                            )) &&
-                        defaultValues[param] &&
-                        looseEquals(value[param], defaultValues[param])
-                    )
-                        hashValue[param] = defaultValues[param];
-                    else hasChanges = true;
-                }
+            updates[key] = value;
+            shouldUpdateStore = true;
+        } else if (defaultValue !== undefined) {
+            const defaultForKey = Array.isArray(defaultValue)
+                ? defaultValue
+                : typeof defaultValue === 'object' && defaultValue !== null
+                  ? defaultValue[key]
+                  : defaultValue;
+
+            if (defaultForKey !== undefined) {
```

Comment:
I also changed this part in my PR #196, I wonder what was the reason you changed this?
in mine I noticed, that `syncFromHash` was causing extra search calls, also breaking the loading indicator.

Reply:
In general I made the handling of search params more robust by implementing standardized URL parsing, supports complex data structures via JSON serialization and enhances maintainability with cleaner, more efficient code organization. This particular section improves performance by avoiding multiple reactive updates when setting several default values at once (like the issue you described).